compiler/noirc_frontend/src/hir/comptime/interpreter/builtin.rs

```diff
@@ -2913,10 +2914,24 @@ fn modulus_num_bits(arguments: Vec<(Value, Location)>, location: Location) -> IR
 }
 
 // fn quoted_eq(_first: Quoted, _second: Quoted) -> bool
-fn quoted_eq(arguments: Vec<(Value, Location)>, location: Location) -> IResult<Value> {
-    eq_item(arguments, location, get_quoted)
-}
+fn quoted_eq(
+    interner: &NodeInterner,
+    arguments: Vec<(Value, Location)>,
+    location: Location,
+) -> IResult<Value> {
+    let (self_arg, other_arg) = check_two_arguments(arguments, location)?;
+    let self_arg = get_quoted(self_arg)?;
+    let other_arg = get_quoted(other_arg)?;
+
+    // Comparing tokens one against each other doesn't work in the general case because tokens
+    // might be refer to interned expressions/statements/etc. We'd need to convert those nodes
+    // to tokens and compare the final result, but comparing their string representation works
+    // equally well and, for simplicity, that's what we do here.
+    let self_string = tokens_to_string(&self_arg, interner);
+    let other_string = tokens_to_string(&other_arg, interner);
```

Comment:
Should we change the Quoted Eq implementation as well for consistency?

Or keep a separation between the type within Noir and what we expose to the comptime API?

Reply:
Ah, good point. The issue is that we need an `interner` in order to compare them so I'm not sure how that can be done in `Eq`.

Reply:
Ah, ok. It's probably fine to have them separate like this.